fl4health/feature_alignment/tab_features_preprocessor.py

```diff
@@ -0,0 +1,147 @@
+
+    def get_one_hot_pipeline(self, categories: MetaData) -> Pipeline:
+        return Pipeline(steps=[("encoder", OneHotEncoder(handle_unknown="ignore", categories=[categories]))])
+
+    def get_ordinal_pipeline(self, categories: MetaData) -> Pipeline:
+        return Pipeline(
+            steps=[
+                (
+                    "encoder",
+                    OrdinalEncoder(
+                        unknown_value=len(categories) + 1,
+                        handle_unknown="use_encoded_value",
+                        categories=[categories],
+                    ),
+                )
+            ]
+        )
+
+    def get_default_string_pipeline(self, vocabulary: MetaData) -> Pipeline:
+        return Pipeline(steps=[("vectorizer", TextColumnTransformer(TfidfVectorizer(vocabulary=vocabulary)))])
+
+    def initialize_default_pipelines(
+        self, tabular_features: List[TabularFeature], feature_to_transformer: Dict[str, Pipeline], one_hot: bool
+    ) -> None:
+        """
+        Initialize a default Pipeline for every data column in tabular_features.
+
+        Parameters
+        ----------
+            tabular_features (List[TabularFeature]): list of tabular
+            features in the data columns.
+        """
+        for tab_feature in tabular_features:
+            feature_type = tab_feature.get_feature_type()
+            feature_name = tab_feature.get_feature_name()
+            if feature_type == TabularType.NUMERIC:
+                feature_pipeline = self.get_default_numeric_pipeline()
+            elif feature_type == TabularType.BINARY:
+                feature_pipeline = self.get_default_binary_pipeline()
+            elif feature_type == TabularType.ORDINAL:
+                feature_categories = tab_feature.get_metadata()
+                if one_hot:
+                    feature_pipeline = self.get_one_hot_pipeline(feature_categories)
+                else:
+                    feature_pipeline = self.get_ordinal_pipeline(feature_categories)
+            else:
+                vocabulary = tab_feature.get_metadata()
+                feature_pipeline = self.get_default_string_pipeline(vocabulary)
+            feature_to_transformer[feature_name] = feature_pipeline
+
+    def return_column_transformer(self, pipelines: Dict[str, Pipeline]) -> ColumnTransformer:
+        transformers = [
+            (f"{feature_name}_pipeline", pipelines[feature_name], [feature_name])
+            for feature_name in sorted(pipelines.keys())
+        ]
+        return ColumnTransformer(
+            transformers=transformers,
+            remainder="drop",
+        )
+
+    def set_feature_pipeline(self, feature_name: str, pipeline: Pipeline, target: bool) -> None:
+        # This method allows the user to customize a specific pipeline to be applied to a specific feature.
+        # For example, the user may want to use different scalers for two distinct numerical features.
+        if not target:
```

Comment:
Just a thought and you don't have to use it, but rather than have the user specify whether the feature is a target or not, could we simply just do
```
if feature_name in self.features_to_pipelines:
    self.features_to_pipelines[feature_name] = pipeline
    self.data_column_transformer = self.return_column_transformer(self.features_to_pipelines)
elif feature_name in self.targets_to_pipelines:
    self.targets_to_pipelines[feature_name] = pipeline
    self.target_column_transformer = self.return_column_transformer(self.targets_to_pipelines)
else:
    log(WARNING, f"{feature_name} is not in either the feature or target pipelines and the provided pipeline will be ignored."
```